src/unix/network_helper.rs

```diff
@@ -116,3 +126,261 @@ pub(crate) fn get_interface_address() -> Result<InterfaceAddressIterator, String
         }
     }
 }
+
+pub(crate) fn get_interface_ip_networks() -> HashMap<String, HashSet<IpNetwork>> {
```

Comment:
I think it'd be better to mark the entire function as unsafe: `unsafe` isn't only limited to a C call or a pointer access but "lives" until the unsafe data has been converted into Rust so to speak. So having local `unsafe` blocks doesn't make much sense.

Reply:
Now I try to add a inner function for unsafe code.

Reply:
Please make the function itself unsafe, don't wrap, it's giving the impression that the function is safe although it's not. ^^'

Reply:
in `unix/network_helper.rs`
```
pub(crate) fn get_interface_ip_networks() -> HashMap<String, HashSet<IpNetwork>> {
    unsafe { inner_interface_ip_networks() }
}

unsafe fn inner_interface_ip_networks() -> HashMap<String, HashSet<IpNetwork>> {
```
however, in `windows/network_helper.rs`

there is another block code like 
```
pub(crate) fn get_interface_ip_networks() -> HashMap<String, HashSet<IpNetwork>> {
    match get_interface_address() {
        Ok(mut interface_iter) => interface_iter.generate_ip_networks(),
        _ => HashMap::new(),
    }
}
```

I don`t know if it will be conflict or confused when call it on different platforms?

Reply:
@GuillaumeGomez I will need more advice about how balance the usage and for different platforms